### PR TITLE
use expected instead of panic

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

File: src/jsonpath/json_path.rs [hunk 1]
```diff
@@ -45,22 +45,23 @@ impl<'i> Query<'i> {
...
-                    let first_on_list = last.into_inner().next();
-                    first_on_list.map(|first| (first.as_str().to_string(), JsonPathToken::String))
-                }
-                _ => panic!("pop last was used in a none static path"),
-            },
-            None => None,
+        let last = self
+            .root
+            .next_back()
+            .expect("pop last was used in a none static path");
```
Comment (oshadmi) [inline src/jsonpath/json_path.rs:51]: When `None` it was returning `None` and not panicking?
(before there was 1 `panic` and now there are `expect` and `panic`)

Comment (gkorland) [reply src/jsonpath/json_path.rs]: @oshadmi good catch not sure why I did it

File: src/formatter.rs [hunk 2]
```diff
@@ -175,6 +175,7 @@ mod tests {
     use super::*;
 
     #[test]
+    #[allow(clippy::cognitive_complexity)]
```
Comment (oshadmi) [inline src/formatter.rs:178]: We are linting cognitive_complexity in tests?

**Review by oshadmi — APPROVED** [review]
LGTM!

File: src/formatter.rs (hunk 2 above)
Comment (gkorland) [reply src/formatter.rs:178]: clippy does

**Pull request merged**